ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -504,6 +594,99 @@ def _undo_encoding(
+        latest_encoded_columns = list(chain(*(i if isinstance(i, list) else (i,) for i in flattened_modes)))
+        # check for duplicates and raise an error if any
+        if len(set(latest_encoded_columns)) != len(latest_encoded_columns):
+            print(
+                "[bold red]Reencoding AnnData object failed. You have at least one duplicate in your encodings. A column "
+                "cannot be encoded at the same time using different encoding modes!"
+            )
+            raise DuplicateColumnEncodingError
+        old_encode_mode = adata.uns["current_encodings"]
+        for categorical in latest_encoded_columns:
+            encode_mode = old_encode_mode.get(categorical)
+            # if None, this categorical has not been encoded before but will be encoded now for the first time
+            # multi column encoding mode
+            if encode_mode in Encoder.multi_encoding_modes:
+                encoded_categoricals_with_mode = adata.uns["categoricals_encoded_with_mode"][encode_mode]
+                _found = False
+                for column_list in encoded_categoricals_with_mode:
+                    for column_name in column_list:
+                        if column_name == categorical:
+                            column_list.remove(column_name)
+                            _found = True
+                            break
+                    # a categorical can only be encoded once and therefore found once
+                    if _found:
+                        break
+                # filter all lists that are now empty since all variables will be reencoded from this list
+                updated_multi_list = filter(None, encoded_categoricals_with_mode)
+                # if no columns remain that will be encoded with this encode mode, delete this mode from modes as well
+                if not list(updated_multi_list):
+                    del adata.uns["categoricals_encoded_with_mode"][encode_mode]
+            # single column encoding mode
+            elif encode_mode in Encoder.available_encodings:
+                encoded_categoricals_with_mode = adata.uns["categoricals_encoded_with_mode"][encode_mode]
+                for ind, column_name in enumerate(encoded_categoricals_with_mode):
+                    if column_name == categorical:
+                        del encoded_categoricals_with_mode[ind]
+                        break
+                    # if encoding mode is
+                if not encoded_categoricals_with_mode:
+                    del adata.uns["categoricals_encoded_with_mode"][encode_mode]
+        # return updated encodings
+        return Encoder._update_new_encode_modes(new_encodings, adata.uns["categoricals_encoded_with_mode"])
+
+    @staticmethod
+    def _update_new_encode_modes(
+        new_encodings: Dict[str, Union[List[List[str]], List[str]]],
+        filtered_old_encodings: Dict[str, Union[List[List[str]], List[str]]],
+    ):
+        """Update the encoding scheme.
```

Comment:
The types look so complex that this function warrants a proper docstring, please.